store/root.js

```diff
@@ -114,16 +114,32 @@ class Root extends Store {
     // since we do not really know what type of ID the API exposes
     if (this.dataProvider?.id === dataProviderInt) return
 
-    const dataProviderMetaData = this.user.getDataProviderById(dataProviderInt)
+    let dataProviderMetaData = this.user.getDataProviderById(dataProviderInt)
 
     // Check access rights
     if (
       dataProviderMetaData !== null &&
       !this.user.canManage(dataProviderMetaData?.id)
     ) {
-      // TODO: allow access to any dataProvider for admin users
-      const dpStr = `DataProvider#${dataProviderMetaData?.id}`
-      throw new AccessError(`${this.user} does not have access to the ${dpStr}`)
+      if (!this.user.superAdmin) {
```

Comment:
This should be on a different level. A bit above ☝️

`User.canManage(id)` should return `true` for any ID if the user is a super-admin. The rest of the code should stay generic.

Reply:
> This should be on a different level. A bit above ☝️

So in that case `getDataProviderById` should fetch dataProvider metadata if user is superAdmin?

> The rest of the code should stay generic.

What do you mean by that?

Reply:
> So, in that case, `getDataProviderById` should fetch dataProvider metadata if user is `superAdmin` and data is missing?

I mean, that the if there is and `/data-providers/{id}`, the API call for the metadata goes anyway. The user who opens the URL most likely has an access to it. The `/user` call goes asynchronously too.

When both are retrieved, we can check the `canManage` condition.

`getDataProviderById` perhaps can be removed.

Reply:
There is no point in rerequesting the data we already have.

Reply:
Let's say, there should be a method that creates a DataProvider store from the ID. Calling this method means that the user can manage the store.

The method itself can find a data provider in the user entity or retrieve it from the server, hence it should be async.

Reply:
If I understood you correctly basically the code I added should be just moved to a new method but the whole logic remains the same.

Reply:
The `canManage` method should change to the following:

```js
  canManage(dataProviderId) {
    return this.superAdmin || this.dataProviders.some(({ id }) => dataProviderId === id)
  }
```

and you should utilise it within the Root store without relying on specific knowledge about the entities like having a `superAdmin` property in the User.

This particular `if` statement should not exist but the `changeDataProvider` method should be updated appropriately considering that some data can be missing.

Also, I don't think the current data provider should be added to the user's list but it should be shown somehow in the menu (as the first item, perhaps).

Reply:
Fair enough. I would suggest merging this PR as I don't see the things you are proposing as a blocker whatsoever. If you feel like it can be improved in any way feel free to refactor it. What do you think would that work for you? 🙂

Reply:
Yeah, both options have pros and cons. I lost error reporting for 404 at `/data-providers/{id}`